### ISSUE TITLE
"bulkwalk" does not work properly

### ISSUE THREAD

**Issue by exhuma** [exhuma/puresnmp#22]
In the current release (1.1.1), bulkwalk raises the following error on OID `1.3.6.1.2.1.2.1`:

```
Traceback (most recent call last):
  File "my__.tester.py", line 90, in <module>
    test_if_oper_stat()
  File "my__.tester.py", line 85, in test_if_oper_stat
    for row in iftable:
  File "/home/users/michel/work/pynet/env/lib/python3.4/site-packages/puresnmp/__init__.py", line 475, in bulkwalk
    for oid, value in result:
  File "/home/users/michel/work/pynet/env/lib/python3.4/site-packages/puresnmp/__init__.py", line 225, in multiwalk
    unfinished_oids = get_unfinished_walk_oids(varbinds, requested_oids)
  File "/home/users/michel/work/pynet/env/lib/python3.4/site-packages/puresnmp/__init__.py", line 201, in get_unfinished_walk_oids
    for k, v in results.items()}
  File "/home/users/michel/work/pynet/env/lib/python3.4/site-packages/puresnmp/__init__.py", line 201, in <dictcomp>
    for k, v in results.items()}
IndexError: list index out of range
```

**Comment by exhuma** [comment]
Seems to be working on my development machine. Will need to check again at the office where this comes from, and more importantly which response packet causes the error.

**Comment by exhuma** [comment]
This happens if `puresnmp.bulkwalk` is called with a string as OID instead of a list of strings. This is related to #24 and will be fixed for release 2.0 (as #24 will break backwards compatibility).

In the mean-time, a type check on `bulkwalk` will be added to catch the error early and give a more helpful message.

**Comment by exhuma** [comment]
This has been fixed in ba202599
